Create SettingsImportScreen UI component

bernardodiasc opened the pull request:
**Release Type:** *Non-Breaking Feature*

Fixes https://x-team-internal.atlassian.net/browse/XP-2991

## Description

Create new `SettingsImportScreen` UI component

## Checklist

- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `IN REVIEW` column
- [x] fill out `Is a breaking change` and `Release Type Reason` (if required) fields for the ticket. This is accessible when editing an issue in JIRA
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`)
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases or variants usage
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot
- [x] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Related PRs

branch | PR
------ | ------
`XP-2991-create-import-profiles-screen` | [XP](https://github.com/x-team/xp/pull/1874)

## Steps to Test or Reproduce

- run `npm start`
- go to http://localhost:9001/?selectedKind=UI%20Components%2FSettingsImportScreen

## Screenshots

![2019-04-19 14 02 40](https://user-images.githubusercontent.com/131859/56435034-57001c00-62ad-11e9-98ee-ba3b31b8b903.gif)

nicksp commented on src/components/ui/SettingsImportScreen.js:
```diff
@@ -0,0 +1,194 @@
+  validEmails: Array<*>,
+  invalidEmails: Array<*>,
+  response: {
+    validProfileEmails: Array<*>,
```
these names look much better now, thanks for applying suggestions 👍

nicksp commented on src/components/ui/SettingsImportScreen.js:
```diff
@@ -0,0 +1,194 @@
+  error: string,
+  handleListSelectionChange: Function,
+  handleListOfEmailsChange: Function,
+  handleImport: Function,
```
Is it possible to use more specific types instead of unsafe and generic `Function`s? I mean, since we know signatures of methods, could we reuse this info as the type?

nicksp commented on src/components/ui/SettingsImportScreen.js:
```diff
@@ -0,0 +1,194 @@
+          onChange={handleListOfEmailsChange}
+          defaultValue={emailsList}
+        />
+        {status === 'VALIDATED' && size(invalidEmails) > 0 && (
```
I'd also suggest to move these hardcoded strings like `VALIDATED`, `VALIDATING` as enums in constants, smth into `IMPORT_EMAILS_STATES` or smth, so it's easier to change in future and easier to understand what they refer to.

nicksp commented:
Great work on isolating these styles and logic into UI component, I like that 👏

bernardodiasc commented:
Thanks for the feedback @nicksp, great points, as usual 😄 I update the code already.

bernardodiasc commented:
@migueliriano interesting, thanks for catching that. The `id === 0` had this intention but when I refactored the code I replaced the `selectedListId` with the `selectedList` that contains the `id`, so maybe we don't need `id === 0` anymore. Anyway, I think the update you made are pretty fine, thanks!
